warehouse/models/mart/gtfs/fct_observed_trips.sql

```diff
@@ -1,41 +1,88 @@
+    SELECT * FROM {{ ref('dim_provider_gtfs_data') }}
+),
+trip_updates AS (
     SELECT * FROM {{ ref('int_gtfs_rt__trip_updates_summaries') }}
 ),
-
 vehicle_positions AS (
     SELECT * FROM {{ ref('int_gtfs_rt__vehicle_positions_trip_summaries') }}
 ),
-
 service_alerts AS (
     SELECT * FROM {{ ref('int_gtfs_rt__service_alerts_trip_summaries') }}
 ),
+--
+-- -- in RT world can use associated_schedule_gtfs_dataset_key to identify "groups" of RT feeds
+-- url_to_associated_schedule_key AS (
+--     SELECT
+--         ff.date,
+--         ff.gtfs_dataset_key,
+--         pgd.associated_schedule_gtfs_dataset_key
+--     FROM fct_daily_rt_feed_files AS ff
+--     LEFT JOIN dim_provider_gtfs_data AS pgd
+--         ON ff.date BETWEEN pgd._valid_from AND pgd._valid_to
+--         AND ff.schedule_to_use_for_rt_validation_gtfs_dataset_key = pgd.associated_schedule_gtfs_dataset_key
+-- ),
+
+-- Per usual, see https://gtfs.org/realtime/reference/#message-tripdescriptor for what uniquely
+-- identifies a trip in a given feed. Computing this ahead of time simplifies the join conditions
+-- since we want to treat "nulls" as joinable.
+trip_updates_with_associated_schedule AS (
+    SELECT
+        trip_updates.*,
+        {{ dbt_utils.surrogate_key(['trip_id', 'trip_route_id', 'trip_direction_id', 'trip_start_time', 'trip_start_date']) }} AS trip_identifier,
+        fct_daily_rt_feed_files.schedule_to_use_for_rt_validation_gtfs_dataset_key AS associated_schedule_gtfs_dataset_key,
+    FROM trip_updates
+    LEFT JOIN fct_daily_rt_feed_files
+        ON trip_updates.dt = fct_daily_rt_feed_files.date
+        AND trip_updates.base64_url = fct_daily_rt_feed_files.base64_url
+),
+vehicle_positions_with_associated_schedule AS (
+    SELECT
+        vehicle_positions.*,
+        {{ dbt_utils.surrogate_key(['trip_id', 'trip_route_id', 'trip_direction_id', 'trip_start_time', 'trip_start_date']) }} AS trip_identifier,
+        fct_daily_rt_feed_files.schedule_to_use_for_rt_validation_gtfs_dataset_key AS associated_schedule_gtfs_dataset_key,
+    FROM vehicle_positions
+    LEFT JOIN fct_daily_rt_feed_files
+        ON vehicle_positions.dt = fct_daily_rt_feed_files.date
+        AND vehicle_positions.base64_url = fct_daily_rt_feed_files.base64_url
+),
+service_alerts_with_associated_schedule AS (
+    SELECT
+        service_alerts.*,
+        {{ dbt_utils.surrogate_key(['trip_id', 'trip_route_id', 'trip_direction_id', 'trip_start_time', 'trip_start_date']) }} AS trip_identifier,
+        fct_daily_rt_feed_files.schedule_to_use_for_rt_validation_gtfs_dataset_key AS associated_schedule_gtfs_dataset_key,
+    FROM service_alerts
+    LEFT JOIN fct_daily_rt_feed_files
+        ON service_alerts.dt = fct_daily_rt_feed_files.date
+        AND service_alerts.base64_url = fct_daily_rt_feed_files.base64_url
+),
 
 fct_observed_trips AS (
     SELECT
         {{ dbt_utils.surrogate_key([
             'dt',
             'associated_schedule_gtfs_dataset_key',
```

Comment:
i think it might be cleaner to use the gtfs_dataset_key, rather than the schedule key

Reply:
Because he's trying to join across separate datasets (trip updates / vehicle positions / service alerts), there are a few things that they have in common:
* `associated_schedule_gtfs_dataset_key` -- this is the schedule dataset that they use for validation (what Andrew is currently using)
* `service` / `organization` (traversing farther through the tree of associations)

I told him to use `associated_schedule_gtfs_dataset_key` because it's the "closest" -- it uses only datasets rather than going out to service/organization.

But these records do not have`gtfs_dataset_key` itself in common, because they are different feeds (trip updates vs. vehicle positions means different `gtfs_dataset_key`, different `base64_url`, etc.). So `associated_schedule_gtfs_dataset_key` is the "nearest" thing that these records would have in common.

Reply:
Of course of course, sorry to make you explain again! I keep going back and forth on remembering what gtfs_dataset_key means, I keep thinking it means "quartet key".

Reply:
Yeah, it is definitely confusing -- as soon as I finish #2204 documentation is my next priority 😅

Reply:
Thanks! Going to resolve this and put this up for review in a sec.